notebooks/trends/trends-notebook.ipynb

```diff
@@ -0,0 +1,553 @@
+    "        forecast_processed_data[\"so2\"][\"values\"].append(forecast[\"so2\"][\"value\"])\n",
+    "        forecast_processed_data[\"so2\"][\"times\"].append(forecast_date)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Select Measuring stations to display\n",
+    "\n",
+    "The list of stations is derived from the data obtained from the API call. \n",
+    "\n",
+    "**ATTENTION: Some of the stations may not have data for a given pollutant. Therefore if you deselect all stations that have data for a given pollutant that pollutants graph will no longer render.**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "stations_to_display = []\n",
+    "\n",
+    "stations_with_duplicates = []\n",
+    "\n",
+    "def update_stations_to_display(stations_input):\n",
+    "      global stations_to_display\n",
+    "      stations_to_display = stations_input\n",
+    "\n",
+    "for pollutant in in_situ_processed_data:\n",
+    "    for measuring_station, station_data in in_situ_processed_data[pollutant].items():\n",
+    "        stations_with_duplicates.append(measuring_station)\n",
+    "\n",
+    "stations = list(set(stations_with_duplicates))\n",
+    "\n",
+    "stations_widget = widgets.SelectMultiple(\n",
+    "    options=stations,\n",
+    "    value=stations,\n",
+    "    description='Stations:',\n",
+    ")\n",
+    "\n",
+    "interact(update_stations_to_display, stations_input=stations_widget)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Generating the Graphs\n",
+    "\n",
+    "The below code cell will generate five graphs. One for each pollutant. Each line represents a measuring station. \n",
+    "\n",
+    "You are also able to change the transparency of the AQI background of the graphs by adjusting the `aqi_block_color_alpha` below. 0.3 is the default.\n",
+    "Furthermore, by default if there are more than 20 stations the legend is hidden, this can be changed by modifying the variable `hide_legend_if_exceed_stations`.\n",
+    "\n",
+    "**WARNING: Below, changing the line `ax.xaxis.set_major_locator(mdates.DayLocator())` to `ax.xaxis.set_major_locator(mdates.HourLocator())` will make the tick labels display every hour instead of every day, this can cause issues though so this is not recommended**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import matplotlib.pyplot as plt\n",
+    "import matplotlib.dates as mdates\n",
+    "import numpy as np\n",
+    "\n",
+    "aqi_block_color_alpha = 0.3\n",
+    "hide_legend_if_exceed_stations = 20\n",
+    "\n",
+    "aqi_ranges = {\n",
+    "    \"o3\": [\n",
+    "         50,\n",
+    "         100,\n",
+    "         130,\n",
+    "         240,\n",
+    "         380,\n",
+    "         800,\n",
+    "    ],\n",
```

Comment:
I'd single line all of these arrays for space saving. Not sure you gain anything by having each element on a new line
```suggestion
    "    \"o3\": [ 50, 100, 130, 240 380 800 ],\n",
```